code-challenges.test.js

```diff
@@ -14,60 +21,137 @@
+// The next step is to sort the values in ascending order. To do this, we use the sort() method on the values array. We pass a comparison function (a, b) => a - b to the sort() method.
+// The comparison function (a, b) => a - b is responsible for determining the order of the elements during the sorting process. It compares two elements, a and b, and returns a negative value if a should come before b, a positive value if a should come after b, or zero if they have the same order.
+// In our case, since we want to sort the values in ascending order, we subtract b from a in the comparison function (a, b) => a - b. This ensures that smaller values come before larger values.
+// Then we return the sorted values array from the function.
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+describe('accuSum', () => {
+  it('takes in an array and returns an array of the accumulating sum. An empty array should return an empty array', () => {
+    const accountTransactions1 = [100, -17, -23, -9]
+    expect(accuSum(accountTransactions1)).toEqual([100, 83, 60, 51])
+
+    const accountTransactions2 = [250, -89, 100, -96]
+    expect(accuSum(accountTransactions2)).toEqual([250, 161, 261, 165])
+
+    const accountTransactions3 = [];
+    expect(accuSum(accountTransactions3)).toEqual([])
+  })
+})
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
 
-const accountTransactions3 = []
-// Expected output: []
+// const accountTransactions1 = [100, -17, -23, -9]
+// // Expected output: [100, 83, 60, 51]
+
+// const accountTransactions2 = [250, -89, 100, -96]
+// // Expected output: [250, 161, 261, 165]
+
+// const accountTransactions3 = []
+// // Expected output: []
 
 // b) Create the function that makes the test pass.
 
+function accuSum(arr) {
+  let sum = 0
+  return arr.map(num => {
+    sum += num
+    return sum
+  })
+}
+
+
 // Pseudo code:
+// First, I define a function called accuSum that takes an array as a parameter. I'll call this parameter arr.
+// Inside the function, I declare a variable called sum and set it to 0. This variable will keep track of the accumulating sum.
+// Next, I use the map() method on the arr array. The map() method iterates over each element of the array and returns a new array with the transformed values.
+// For each element in the arr array, I define an arrow function that takes a parameter called num. This arrow function represents the transformation to be applied to each element.
+// Inside the arrow function, I update the sum variable by adding the current element num to it.
+// Finally, I return the updated sum after each iteration, which gives us the accumulating sum at each position in the array.
```

Comment:
Nice job. You have great explanations for these solutions in your pseudo code. Make sure you are using arrow functions instead of the function keyword to declare functions. There are differences between the two pertaining to their `this` scopes, but you can research that on your own if you wish.